pkg/reidentification/distance.go

```diff
@@ -0,0 +1,99 @@
+package reidentification
+
+import (
+	"math"
+	"sort"
+)
+
+type Cosine struct{}
+
+type Euclidean struct{}
+
+type Manhattan struct{}
+
+type Canberra struct{}
+
+type Chebyshev struct{}
+
+type Minkowski struct {
+	p float64
+}
+
```

Comment:
Struct is overkill for distance function !  just define a  `CosineDistance` function.